### PR TITLE
Add matching on image ID and version

### PR DESCRIPTION
**Pull request by behrmann**

### CODE REVIEW

File: mkosi.md [hunk 1]
```diff
@@ -234,6 +234,14 @@ a boolean argument: either "1", "yes", or "true" to enable, or "0",
   interpreted relative to the parent directory of the config file that the
   condition is read from.
 
+`ImageId=`
+
+: Matches against the configured image ID. If this condition is used and no
+  image ID has been explicitly configured yet, this condition fails. Multiple
+  image IDs may be specified, separated by spaces. If multiple releases are
```
Comment (DaanDeMeyer) [inline mkosi.md:241]: releases => image ids

**Review by DaanDeMeyer — APPROVED** [review]
`ImageVersion=` matcher is very nice to read. Needs a CI fix though.

**Comment by behrmann** [comment]
Ah... `systemd-analyze` doesn't seem to be picked up in the tests
```
2023-04-24T20:45:23.3325658Z Unknown command verb compare-versions.
2023-04-24T20:45:23.3387639Z Unknown command verb compare-versions.
2023-04-24T20:45:23.3451192Z Unknown command verb compare-versions.
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -17,6 +17,23 @@ jobs:
...
+    shell: bash
+    working-directory: ${{ github.action_path }}
+    run: |
+      echo "deb-src http://archive.ubuntu.com/ubuntu/ $(lsb_release -cs) main restricted universe multiverse" | sudo tee -a /etc/apt/sources.list
+      sudo apt-get update
+      sudo apt-get build-dep systemd
+      sudo apt-get install libfdisk-dev
+
+      git clone https://github.com/systemd/systemd --depth=1
+      meson systemd/build systemd -Drepart=true -Defi=true -Dbootloader=true
```
Comment (DaanDeMeyer) [inline .github/workflows/ci.yml:30]: Let's drop the meson options since they're not needed for analyze

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -17,6 +17,23 @@ jobs:
...
     - uses: actions/checkout@v3
 
+  - name: Update systemd-analyze
+    shell: bash
+    working-directory: ${{ github.action_path }}
+    run: |
+      echo "deb-src http://archive.ubuntu.com/ubuntu/ $(lsb_release -cs) main restricted universe multiverse" | sudo tee -a /etc/apt/sources.list
+      sudo apt-get update
+      sudo apt-get build-dep systemd
+      sudo apt-get install libfdisk-dev
```
Comment (DaanDeMeyer) [inline .github/workflows/ci.yml:27]: Shouldn't need libfdisk-dev for systemd-analyze

**Review by DaanDeMeyer — APPROVED** [review]

**Pull request merged**